libquery/src/CypherCompiler.cpp

```
@@ -1046,7 +1046,10 @@ galois::CypherCompiler::CompileWithOrReturn(CypherASTNode ast, bool is_return) {
     this->projection_metadata.distinct = cypher_ast_with_is_distinct(ast);
   }
 
-  this->projection_metadata.order_by = ParseOrderByNode(ast, is_return);
+  auto res = ParseOrderByNode(ast, is_return);
+  if (res) {
```

should we propagate an error if `ParseOrderByNode` returns an error?

---

The old version had 3 output possibilities 1) success, non-null 2) return nullptr 3) crash
I combined 2 and 3, I guess I should be keeping all of these